source/azure_iot_hub_client.c

```diff
@@ -346,6 +355,95 @@ static uint32_t prvGetTimeMs( void )
 }
 /*-----------------------------------------------------------*/
 
+static AzureIoTHubClientResult_t prvRequestIdGet( AzureIoTHubClient_t * pxAzureIoTHubClient,
+                                                  az_span xSpan,
+                                                  bool xOddSeq,
+                                                  uint32_t * pulRequestId,
+                                                  az_span * pxSpan )
+{
+    AzureIoTHubClientResult_t xReturn;
+    az_span xRemainder;
+    az_result xCoreResult;
+
+    if( xOddSeq == ( bool ) ( pxAzureIoTHubClient->_internal.ulCurrentRequestID & 0x01 ) )
+    {
+        pxAzureIoTHubClient->_internal.ulCurrentRequestID += 2;
+    }
+    else
+    {
+        pxAzureIoTHubClient->_internal.ulCurrentRequestID += 1;
+    }
+
+    if( pxAzureIoTHubClient->_internal.ulCurrentRequestID == 0 )
+    {
+        pxAzureIoTHubClient->_internal.ulCurrentRequestID = 2;
+    }
+
+    xCoreResult = az_span_u32toa( xSpan, pxAzureIoTHubClient->_internal.ulCurrentRequestID, &xRemainder );
+
+    if( az_result_failed( xCoreResult ) )
+    {
+        AZLogError( ( "Couldn't convert request id to span" ) );
+        xReturn = eAzureIoTHubClientFailed;
+    }
+    else
+    {
+        *pxSpan = az_span_slice( *pxSpan, 0, az_span_size( *pxSpan ) - az_span_size( xRemainder ) );
+
+        if( pulRequestId )
+        {
+            *pulRequestId = pxAzureIoTHubClient->_internal.ulCurrentRequestID;
+        }
+
+        xReturn = eAzureIoTHubClientSuccess;
+    }
+
+    return xReturn;
+}
+/*-----------------------------------------------------------*/
+
+static AzureIoTHubClientResult_t prvWaitForSubAck( AzureIoTHubClient_t * pxAzureIoTHubClient,
```

Comment:
Nit - consider a LogDebug (I think that's right level) for when we enter and if we leave this function successfully.  Reason I suggest is that since we could conceivable be blocking for a bit during a SUBACK (as these are rare and we force caller to wait here, which is fine) there could be questions about a thread's lifeliness if we're gone "too long".  

I don't think we need/want per-ProcessLoop() as that might get to chatty (or that's some crazy packet level thing whic is probably diferent discussion).  Just the entry/exit.